pkg/controller/replicaset/replica_set_test.go

```diff
@@ -1151,6 +1151,7 @@ func TestExpectationsOnRecreate(t *testing.T) {
 		100,
 	)
 	f.Start(stopCh)
+	f.WaitForCacheSync(stopCh)
 	fakePodControl := controller.FakePodControl{}
```

Comment:
thanks… this is the second time an issue has come up like this in a unit test. do you mind sweeping for other places we're setting up NewSharedInformerFactory in unit tests to see if there are other places we need to do this?

Reply:
@liggitt I would love to and I wrote a small program https://gist.github.com/knight42/73c13a0166a5d0884545f78bafc64acc to find out all the cases, and we seem to be lucky:

```
$ ./detect pkg
pkg/controller/replicaset/replica_set_test.go TestWatchControllers
pkg/controller/replicaset/replica_set_test.go TestExpectationsOnRecreate
$ ./detect staging
```

Note that there might be false negatives, i.e. if sharedInformerFactory is not created or started in test function directly, the program would fail to detect.

Reply:
great, thanks! do you mind adding the waits to those tests as well in this PR?